### PR TITLE
Implement test case for joining two tables with identical column names

### PR DESCRIPTION
**Pull request by rozenmd**

### CODE REVIEW

**Comment by rozenmd** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/workerd/api/sql-test.js [hunk 1]
```diff
@@ -26,6 +26,27 @@ async function test(storage) {
...
+  INSERT INTO users VALUES (1, 'Alice');
+  INSERT INTO users VALUES (2, 'Bob');
+  INSERT INTO articles VALUES (1, 'Hello World', 1);
+  INSERT INTO articles VALUES (2, 'Goodbye World', 2);`);
+  {
+    let result = sql.exec(`
+    SELECT articles.name, users.name
+    FROM articles
+    INNER JOIN users
+      ON articles.user_id = users.id`).raw();
```
Comment (kentonv) [inline src/workerd/api/sql-test.js:42]: This returns an iterator, but the code below is expecting an array. You need something like `let array = [...iter];` to expand it.

**Review by kentonv — COMMENTED** [review]
I guess this test is just to prove that the bug is in D1, not workerd?

**Comment by rozenmd** [comment]
Yeah I wasn't sure where the bug was coming from, so figured this would be a good test to have

**Comment by geelen** [comment]
Closing this in favour of #696

**Pull request closed without merging**